plaso/parsers/postgresql.py

```diff
@@ -0,0 +1,190 @@
+      pyparsing.ZeroOrMore(pyparsing.lineEnd())
+  )
+
+  LINE_STRUCTURES = [('logline', _LOG_LINE)]
+  _SUPPORTED_KEYS = frozenset([key for key, _ in LINE_STRUCTURES])
+
+  def _ConvertTimeString(self, structure):
+    """Converts the structure to a datetime object.
+
+    The date and time values are formatted as:
+    "2022-04-12 00:16:05.526 UTC".
+
+    Args:
+      structure (pyparsing.ParseResults): structure of tokens derived from a
+          line of a text file, that contains the time elements.
+
+    Returns:
+      datetime: datetime object.
+
+    Raises:
+      ValueError: if the structure cannot be converted into a datetime.
+    """
+    try:
+      time_string = '{0:d}-{1:d}-{2:d} {3:d}:{4:d}:{5:d}.{6:d} {7:s}'.format(
+          structure['year'], structure['month'], structure['day_of_month'],
+          structure['time']['hours'], structure['time']['minutes'],
+          structure['time']['seconds'], structure.get('microseconds', [0])[0],
+          structure['time_zone'])
+    except (TypeError, ValueError) as exception:
+      raise ValueError(
+          'unable to format date time string with error: {0!s}.'.format(
+              exception))
+    return parser.parse(timestr=time_string)
+
+
+  def ParseRecord(self, parser_mediator, key, structure):
+    """Parses a record and produces a PostgreSQL event.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      key (str): name of the parsed structure.
+      structure (pyparsing.ParseResults): elements parsed from the file.
+
+    Raises:
+      ParseError: when the structure type is unknown.
+    """
+    if key not in self._SUPPORTED_KEYS:
+      raise errors.ParseError(
+          'Unable to parse record, unknown structure: {0:s}'.format(key))
+
+    event_data = PostgreSQLEventData()
+    event_data.pid = ''.join(
+        [str(pid) for pid in self._GetValueFromStructure(structure, 'pid')])
+
+    log_level = self._GetValueFromStructure(structure, 'log_level')
+    if log_level and len(log_level) != 1:
+      parser_mediator.ProduceExtractionWarning('no log level found')
+      return
+    event_data.log_level = log_level[0]
+
+    user_and_database = self._GetValueFromStructure(
+        structure, 'user_and_database')
+    if user_and_database:
+      event_data.user = ''.join(user_and_database)
+
+    log_line = self._GetValueFromStructure(structure, 'log_line')
+    if log_line:
+      event_data.log_line = log_line.lstrip().rstrip()
+
+    date_time_structure = self._GetValueFromStructure(structure, 'date_time')
+    date_time_parsed = self._ConvertTimeString(date_time_structure)
+    date_time = dfdatetime_time_elements.TimeElementsInMicroseconds()
+    date_time.CopyFromDatetime(date_time_parsed)
+    event = time_events.DateTimeValuesEvent(
+        date_time, definitions.TIME_DESCRIPTION_RECORDED,
+        date_time_parsed.tzinfo)
```

Comment:
make sure to specify kwargs with the keyword such as `time_zone=date_time_parsed.tzinfo`

Reply:
Done